src/Sylius/Bundle/ApiBundle/Controller/Payment/GetPaymentConfiguration.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller\Payment;
+
+use Sylius\Bundle\ApiBundle\Provider\PaymentConfigurationProviderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Repository\PaymentRepositoryInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class GetPaymentConfiguration
+{
+    /** @var PaymentRepositoryInterface */
+    private $paymentRepository;
+
+    /** @var PaymentConfigurationProviderInterface */
+    private $paymentConfigurationProvider;
+
+    public function __construct(
+        PaymentRepositoryInterface $paymentRepository,
+        PaymentConfigurationProviderInterface $paymentConfigurationProvider
+    ) {
+        $this->paymentRepository = $paymentRepository;
+        $this->paymentConfigurationProvider = $paymentConfigurationProvider;
+    }
+
+    public function __invoke(Request $request): JsonResponse
+    {
+        /** @var PaymentInterface|null $payment */
+        $payment = $this->paymentRepository->find($request->get('paymentId'));
+
+        Assert::notNull($payment);
+
+        return new JsonResponse($this->paymentConfigurationProvider->provide($payment));
```

Comment:
```suggestion
        $payment = $this->paymentRepository->find($request->get('paymentId'));

        Assert::notNull($payment);

        return new JsonResponse($this->paymentConfigurationProvider->provide($payment));
```

This endpoint is opened for all users, login and not. We have access to payment `id` and order `id` and we should check next steps:
1) get user form userContext 
2) check login status
3) get order and assert order owner
4) if user is order owner next we should check if payment is on this order 
If all above conditions are met we can response configuration
Similar partially configuration is here: https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ApiBundle/DataProvider/OrderCollectionDataProvider.php

Reply:
i've created order as a user and i couldnt access it without jwt token, we check if payment belongs to order now.